### ISSUE TITLE
 When turning Arways Run on and off again, the joystick Run button isn't remembered

### ISSUE THREAD

**Issue by Zodomaniac** [chocolate-doom/chocolate-doom#997]
### Background

Version of Chocolate Doom: Latest git

Operating System and version: Windows 7

Game: (Doom/Heretic/Hexen/Strife/other) All

Any loaded WADs and mods (please include full command line):

### Bug description

Observed behavior: When turning Always Run on and off again, the joystick Run button isn't remembered if it isn't the default `joyb_speed 2`.
@fabiangreffrath:
>Hm, there must be an additional layer of indirection that I have missed or not understood yet. No matter which button on my gamepad I assign to "run", it always ends up as `joyb_speed 3` in my default.cfg.

Expected behavior:
Maybe it makes sense to implement the following: when the config is being saved with `joybspeed<20`, back up `joybspeed` to some extra config variable (`joybspeed_backup`, being 2 by default as well), and when disabling Always Run, read the key binding from that backup?

**Comment by fabiangreffrath** [comment]
@fragglet I had a shot at this, but have to admit that the additional `joystick_physical_button` abstraction layer doesn't make things easier to understand. We should probably decide in advance if we want to back up the physical joystick button previously assigned to "Run" or its virtual pendant or both?

**Comment by fragglet** [comment]
Yeah, sorry. The logical/physical button think is very confusing, but I didn't see an easier way of doing it without breaking config file compatibility :(

Given that the `joybspeed` variable is something of a special snowflake, it's probably easiest to preserve the physical button number. You just need to make sure that when you restore it, you only do so if it's not now being used for a different variable.
